### PR TITLE
Parameters Development

### PR DESCRIPTION
**Pull request by StoneyDSP**
Started new Parameters files

### CODE REVIEW

**Comment by StoneyDSP** [comment]
Not finished yet...

**Comment by StoneyDSP** [comment]
An Audio Processor Value Tree State object is instantiated at the master ("PluginProcessor") level and given the name "apvts". Our "apvts" object is returned and passed to "PluginParameters" and "PluginWrapper" via their respective constructors. 

The user interactions with the Parameter objects is calculated within the "PluginParameter" .cpp/.h files. A simple assertion is done at compile done to check that we have assigned the parameter set to our master processor's APVTS object as expected (referenced by name, via string).

The DSP actionis instantiated in the "PluginWrapper" .cpp/.h files, with it's own internal parameters pointing to memory allocations at the Wrapper interval. These allocations are accessed by the APVTS (which in turn is user-controlled via "PluginParameters") via the constructor. Another assertion check is done to make sure we're picking up the expected objects (again reference by name, via string).

The "PluginParameters" files not only interact with the memory objects in "PluginWrapper", but also return a vectorized "setParameterLayout" function, which returns a vector of unique "juce::rangedAudioParameter" objects. This vector of parameter objects is actually instantiated at the Processor level, where it is given the name "params", then passed into "PluginParameter::setParameterLayout", and then return to the processor where it is exposed to the APVTS's parameter layout functions.

In order for all of this to work, a forward declaration/back reference for our derived AudioProcessor class is required in both the Wrapper and Parameter files. This happens to align them nicely with the Editor files in terms of function, and bring a lot of other functions forward.

TL;DR - done.
